packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,157 @@
+        id="name"
+        leadingIcon={EyeIcon}
+        trailingIcon={CloseIcon}
+      />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should be focussed when autoFocus flag is passed', () => {
+    const label = 'Enter name';
+    // eslint-disable-next-line jsx-a11y/no-autofocus
+    const { getByLabelText } = renderWithTheme(<BaseInput label={label} id="name" autoFocus />);
+
+    const input = getByLabelText(label);
+    expect(input).toHaveFocus();
+  });
+
+  it('should be disabled when isDisabled flag is passed', () => {
+    const label = 'Enter name';
+    const { getByLabelText } = renderWithTheme(<BaseInput label={label} id="name" isDisabled />);
+
+    const input = getByLabelText(label);
+    expect(input).toBeDisabled();
+  });
+
+  it('should handle onChange', async () => {
+    const label = 'Enter name';
+    const onChange = jest.fn();
+    const user = userEvent.setup();
+    const userName = 'Divyanshu';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" name="name" onChange={onChange} />,
+    );
+
+    const input = getByLabelText(label);
+    await user.type(input, userName);
+
+    // should be called for each keystroke
+    expect(onChange).toHaveBeenCalledTimes(userName.length);
+    expect(onChange).toHaveBeenLastCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should handle onBlur', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const userName = 'Divyanshu';
+    const onBlur = jest.fn();
+
+    renderWithTheme(
+      <BaseInput
+        label={label}
+        id="name"
+        name="name"
+        defaultValue={userName}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
+        onBlur={onBlur}
+      />,
+    );
+
+    // shifts user focus and therefore blurs the focussed input
+    await user.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
+    expect(onBlur).toHaveBeenCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should set value as an uncontrolled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" defaultValue={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, ' Maithani');
+    expect(input).toHaveValue(valueFinal);
+  });
+
+  it('should set value as a controlled input', () => {
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const { getByLabelText, rerender } = renderWithTheme(
+      <BaseInput label={label} id="name" value={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
```

Comment:
nit: since input is not being used anywhere else

```suggestion
    expect(getByLabelText(label)).toHaveValue(valueInitial);
```

Reply:
Intentionally did it this way to improve readability 😸